packages/core/src/components/BaseDropdown/BaseDropdown.tsx

```diff
@@ -143,7 +142,7 @@ export const HvBaseDropdown = ({
   onContainerCreation,
   ...others
 }: HvBaseDropdownProps) => {
-  const { rootId } = useContext(HvThemeContext);
+  const { rootId } = useTheme();
```

Comment:
This one seems strange coming from a `useTheme` hook (it was already, coming from a `HvThemeContext`). I understand why, and that it is purely theme/styling related, but for consumers it is more something like the app container.

A `useRoot` returning the `id` and possibly a `ref` for the dom element would be interesting.

Reply:
Also, being in separate contexts would also allow changing the theme without triggering re-renders in components using the root element's id/reference, as it most likely never change.